### PR TITLE
DOCS-2361: Autogenerated Felix docs 

### PR DESCRIPTION
**Pull request by ctauchen**
Together with https://github.com/projectcalico/calico/pull/9325, this PR introduces major changes to how our Felix configuration docs are presented.

Major items:
* All data comes from the codebase
* Diffs between codebase and docs are eliminated (plus cross-version, cross-product diffs)
* Process much easier to maintain going forward
* Presentation for users is VASTLY improved.

* [Felix CRD page](https://deploy-preview-1717--calico-docs-preview-next.netlify.app/calico/next/reference/resources/felixconfig/)
* [Felix config/env page](https://deploy-preview-1717--calico-docs-preview-next.netlify.app/calico/next/reference/felix/configuration/)

Product Version(s):

Issue:

Link to docs preview:

SME review:
- [ ] An SME has approved this change. 

DOCS review:
- [ ] A member of the docs team has approved this change.

Additional information:

Merge checklist:

- [ ] Deploy preview inspected wherever changes were made 
- [ ] Build completed successfully
- [ ] Test have passed

### CODE REVIEW

File: src/___new___/components/FelixConfigTable/index.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+                    ) : (
+                      <code>{fieldData.YAMLDefault}</code>
+                    )}
+                  </td>
+      </tr>
+    </tbody>
+  </table>
+);
+
+const TableEnv = ({ fieldData }) => (
```
Comment (ronanc-tigera) [inline src/___new___/components/FelixConfigTable/index.js]: For the sake of neatness, you could extract these components into their own files and import them
```
/FelixConfig
  - index.js
  - file.json
  - TableEnv.js
  - TableConfig.js
  - TableCRD.js

File: src/___new___/components/FelixConfigTable/index.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+      break;
+    case 'env':
+      table = <TableEnv fieldData={fieldData} />;
+      break;
+    default:
+      table = <p>Unknown config type</p>;
+  }
+
+  return (
+    <div>
```
Comment (ronanc-tigera) [inline src/___new___/components/FelixConfigTable/index.js]: no need for the empty div (unless you actually need it). You can do
```
return <>{table}</>

File: src/___new___/components/FelixConfig/index.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+const FelixConfig = ({ configType, name }) => {
+  const matchedGroup = jsonData.Groups.find(group => group.Name === name);
+  // Debugging logs
+
+  if (!matchedGroup) {
+    return <p>No matching group found for '{name}'.</p>;
+  } 
+
+
```
Comment (ronanc-tigera) [inline src/___new___/components/FelixConfig/index.js:37]: make sure you format all of your files `option + shift + f`. You may need to install prettier if it doesn't work

File: src/___new___/components/FelixConfigTable/index.js [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+      </tr>
+    </thead>
+    <tbody>
+      <tr>
+      <td>Key</td>
+      <td><code>{fieldData.NameYAML || 'No Default Value'}</code></td>
+      </tr>
+      <tr>
+        <td>Description</td>
+        <td dangerouslySetInnerHTML={{ __html: fieldData.DescriptionHTML }} />
```
Comment (ronanc-tigera) [inline src/___new___/components/FelixConfigTable/index.js]: You should probably sanitize this html to be safe
https://blog.logrocket.com/using-dangerouslysetinnerhtml-react-application/

Comment (ctauchen) [reply src/___new___/components/FelixConfigTable/index.js]: Done.

File: src/___new___/components/FelixConfig/index.js (hunk 3 above)
Comment (ctauchen) [reply src/___new___/components/FelixConfig/index.js:37]: Done, thanks!

File: src/___new___/components/FelixConfigTable/index.js (hunk 2 above)
Comment (ctauchen) [reply src/___new___/components/FelixConfigTable/index.js]: Ok, will do.

File: src/___new___/components/FelixConfigTable/index.js (hunk 1 above)
Comment (ctauchen) [reply src/___new___/components/FelixConfigTable/index.js]: Done.

File: src/___new___/components/FelixConfig/index.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+          </div>
+        ))}
+      </div>
+    );
+  } else if (configType === 'configenv') {
+    content = (
+      <div>
+        {matchedGroup.Fields.map((field, index) => (
+          <div key={index}>
+            <Tabs groupId='operating-systems'>
```
Comment (ronanc-tigera) [inline src/___new___/components/FelixConfig/index.js]: I noticed all tabs would change at the same time. Not sure if you intended this, but to stop that behaviour remove `groupId='operating-systems'`

File: src/___new___/components/FelixConfig/index.js [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+              <TabItem
+                value='apple'
```
Comment (ronanc-tigera) [inline src/___new___/components/FelixConfig/index.js]: you should probably replace `value='apple/orange'` with something more meaningful like `value='configurationFile/environmentVariable'`

File: src/___new___/components/FelixConfig/index.js (hunk 5 above)
Comment (ctauchen) [reply src/___new___/components/FelixConfig/index.js]: That bit was intentional.

File: src/___new___/components/FelixConfig/index.js (hunk 6 above)
Comment (ctauchen) [reply src/___new___/components/FelixConfig/index.js]: Yes!

**Pull request merged**